src/graph/context/ast/CypherAstContext.h

```diff
@@ -46,7 +46,7 @@ struct NodeInfo {
 
 struct EdgeInfo {
   bool anonymous{false};
-  MatchStepRange* range{nullptr};
```

Comment:
As I known, this AST is released when query executed.

Reply:
Isn't it in the memory pool?

Reply:
All AST will hold by `QueryContext`.

Reply:
AST information will be transferred to Sentence in validate phase.  the sentence is saved in the querycontext during the execution phase,. The current problem is the wild pointer caused by the release of the temporary variable

Reply:
Better change https://github.com/vesoft-inc/nebula/blob/fdca644c7be90719cddfb10cb0735f0b7533136c/src/graph/validator/MatchValidator.cpp#L1180 to `std::vector<MatchPath*>`, we assume lifetime of AST is available before complete query execution.